src/components/dock/dock.types.ts

```diff
@@ -0,0 +1,75 @@
+export interface DockItem {
+    expanded: boolean;
+    useMobileLayout: boolean;
+}
+
+export interface DockItemConfig<T = any> {
+    /**
+     * Text to display for the item.
+     */
+    label: string;
+
+    /**
+     * xxxx
+     */
+    helperLabel?: string;
+
+    /**
+     * True if the list item should be selected.
+     */
+    selected?: boolean;
+
+    /**
+     * Text color of selected Dock item.
+     */
+    selectedTextColor?: string;
+
+    /**
+     * Background color of selected Dock item.
+     */
+    selectedBackgroundColor?: string;
+
+    /**
+     * Value of the Dock item.
+     */
+    value?: T;
```

Comment:
should we call this `id` instead?

Reply:
If an `id` could be useful (for example for aria labelling purposes), we can make it work without a generic value and have `id: string` instead.

But I think a generic value is more useful, with an id you have to lookup the value that you need in your event handler.

Reply:
I changed this to `id: string`, but Bef wanted to keep the option to use a custom type, so he's added `value: T` again. I'm not fully convinced Lime Elements should have to care about what data a consumer wants to "attach" to the dock item, so I'm not sure we should keep the `value` property. I think it might make more sense for the consumer to use the `id` to look up the additional data needed.

Regardless, I'm not a fan of the property name `value` in this instance. It makes sense for things like fields, but a dockItem doesn't really have a "value", in my opinion. This is basically just a way for a consumer to attach some data to the item, that they would like to get back along with the item when the `itemSelected` event is emitted. As such, I think it makes more sense either to have a property like `eventPayload: any`, or simply allow any extra properties on the object by adding `[key: string]: any`.

The problem with specifying an "attachment" like this is that we basically take on responsibility for it. I don't like that at all. It feels very contrary to the SOLID principles to me… 😞

Reply:
`value` has been removed again, and Bef found a way to work around it in crm-components 👌